### DISCUSSION TITLE
What framework are we using?

### DISCUSSION THREAD

**Discussion by AngelCIICMorales** [uprm-inso4101-2023-2024-s2/semester-project-personalfinancemanager#13] [Polls]
We need to figure out which framework to use.

**Comment by diegoq48** [comment]
When doing web development, we use both technologies. Node.js is for running JavaScript on a server (your computer), while React is a library for component-based frontend development. They both utilize JavaScript, and you can use both in your development. In terms of frameworks, it would make more sense to question which JavaScript framework to use, such as React, Next.js, or Angular. Node.js is a runtime tool for running JavaScript outside the browser.

**Comment by Cruiz102** [comment]
Personally, I would prefer the project to be developed with Next.js. The advantages of using Next.js versus a React project include its already structured backend for us. It would be ideal for everyone involved in the project to use JavaScript for both backend and frontend code. In this way, when teams develop their features, they will work on the same codebase, using the same technologies. Many people have recommended using Python with Flask, but that would mean having to install separate dependencies for other programming languages, and the deployment would become a bit more complicated. We could use Docker, but continually adding technologies and dependencies to the project means that every developer should have an understanding of them, especially since we are working on a feature-based approach, meaning that we should work on features from end to end. So, in my opinion, the development should shift to Next.js and more integrated technologies.
